arbnode/dataposter/data_poster.go

```diff
@@ -210,28 +216,73 @@ func (p *DataPoster[Meta]) getFeeAndTipCaps(ctx context.Context, lastTipCap *big
 	return newFeeCap, newTipCap, nil
 }
 
+type SignedBlobTxWrapper struct {
+	types.SignedBlobTx
+}
+
+func (s *SignedBlobTxWrapper) isFake() bool {
+	return false
+}
+
 func (p *DataPoster[Meta]) PostTransaction(ctx context.Context, dataCreatedAt time.Time, nonce uint64, meta Meta, to common.Address, calldata []byte, gasLimit uint64) error {
 	p.mutex.Lock()
 	defer p.mutex.Unlock()
 	feeCap, tipCap, err := p.getFeeAndTipCaps(ctx, nil, dataCreatedAt)
 	if err != nil {
 		return err
 	}
-	inner := types.DynamicFeeTx{
-		Nonce:     nonce,
-		GasTipCap: tipCap,
-		GasFeeCap: feeCap,
-		Gas:       gasLimit,
-		To:        &to,
-		Value:     new(big.Int),
-		Data:      calldata,
+	var txData types.TxData
+	var txWrapData types.TxWrapData
+	if p.isEip4844 {
+		dataBlobs := blobs.EncodeBlobs(calldata)
+		commitments, versionedHashes, aggregatedProof, err := dataBlobs.ComputeCommitmentsAndAggregatedProof()
+		if err != nil {
+			return err
+		}
+		tCap, ok := uint256.FromBig(tipCap)
+		if !ok {
+			return errors.New("tip cap is not a big int")
+		}
+		fCap, ok := uint256.FromBig(feeCap)
+		if !ok {
+			return errors.New("fee cap is not a big int")
+		}
+		txData = &types.SignedBlobTx{
+			Message: types.BlobTxMessage{
+				Nonce:               view.Uint64View(nonce),
+				GasTipCap:           view.Uint256View(*tCap),
+				GasFeeCap:           view.Uint256View(*fCap),
+				Gas:                 view.Uint64View(gasLimit),
+				To:                  types.AddressOptionalSSZ{Address: (*types.AddressSSZ)(&to)},
+				Value:               view.Uint256View(*uint256.NewInt(0)),
+				BlobVersionedHashes: versionedHashes,
+				MaxFeePerDataGas:    view.Uint256View(*fCap), // Use the same fee cap as gas for now.
```

Comment:
We'll need to still have data here. Instead of `isEip4844`, we should have an optional `eip4844Blobs types.Blobs` field which can be nil, and is supplied in addition to the data, not instead of the data.

Reply:
Ah the calldata contains the actual info supplied to the contract and its method, nice catch. Will fix that up soon